Calibration/HcalAlCaRecoProducers/plugins/AlCaDiJetsProducer.cc

```diff
@@ -32,26 +33,36 @@
 #include "Geometry/Records/interface/CaloGeometryRecord.h"
 #include <iostream>
 
+namespace alcaHcalDiJet {
+  struct Counters {
+    Counters() : nAll_(0), nSelect_(0) {}
+    mutable std::atomic<unsigned int> nAll_, nSelect_;
+  };
+}  // namespace alcaHcalDiJet
+
 //
 // class declaration
 //
 
-class AlCaDiJetsProducer : public edm::EDProducer {
+class AlCaDiJetsProducer : public edm::global::EDProducer<edm::RunCache<alcaHcalDiJet::Counters>> {
```

Comment:
why is this prefered to be a `global` instead of a `one` module?

Reply:
According to the [manual](https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes#One_Modules)

> A One module is a module which, for what ever reason, is unable to be used by multiple threads simultaneously and is therefore thread-unsafe. Like a Global module, one instance of a One module is shared across all Streams. However, unlike a Global module, the framework enforces thread-safety by only allowing one Event, LuminosityBlock or Run at a time to be passed to a One module instance. The use of One modules is terrible for performances.

Reply:
Thanks @mmusich I thought that the ALCARECOs are to be produces single-threaded, i.e. as a `one` module, so then this DiJetsProducer is an exception?

Reply:
I think only express is configured to be single-threaded. In Prompt everything is multi-threaded.